pkgs/flox-activation-scripts/default.nix

```diff
@@ -38,7 +38,11 @@ let
     # Note that substitution doesn't work with variables containing "-"
     # so we need to create and use alternative names.
     process_compose = process-compose;
-    flox_activations = flox-activations;
+    flox_activations =
+      if flox-activations != null then
+        "${flox-activations}/bin/flox-activations"
+      else
+        "$FLOX_ACTIVATIONS_BIN";
```

Comment:
blocking: can you run the resulting artifact outside of the devShell where `$FLOX_ACTIVATIONS_BIN` isn't set?

Reply:
If it's unset the activation scripts won't work no. The rust binaries are different since they pick the env var up at _buildtime_ and have the result compiled in so they don't rely on it at runtime.
The equivalent for the bash scripts would be referring to their dependencies by "relative path to themselves" because nix won't be able to know where they will be put and thus can't substitute an absolute path here.

Is running the results _outside of the shell_ (even with `nix develop -c ...`) blocking?

Reply:
I regularly use the devshell to build something and then test it outside of the devshell because certain things don't work inside the devshell.

Reply:
I think people outside the team do so as well

Reply:
well i think the canonical way would be to create a full nix build, using development builds outside of the development environment seems strange.

[6855dc1](https://github.com/flox/flox/pull/2481/commits/6855dc1c28d66ecb73fb17d21c09f3728278e73f) adds some workaround for this situation though.

tl;dr; `./build/wrapper <cmd> <args>` should successfully run `<cmd> <args>` within an environment with the critical values set, e.g. `./build/wrapper flox install hello`

Reply:
Can we use `"${BASH_SOURCE[0]}/../whatever"` here and get rid of the build wrapper?

Reply:
Not without adding multiple variables I think, because the flox activations in particle is called from scripts at different depths. Will check in Monday.

Reply:
`FLOX_ACTIVATIONS_BIN` is currently read by both `activation-scrips/activate` and `activation_scripts/activate.d/{start,attach-inplace}.bash`.
This would mean that we'd need to set it to `${BASH_SOURCE[0]}/../flox-activations/bin/flox-activations` and `${BASH_SOURCE[0]}/../../flox-activations/bin/flox-activations` respectively.

So apart from a wrapper we can, 1) not use `FLOX_ACTIVATIONS_BIN` in one of them, or at least do a better job to pass it to `activate.d/{start,attach-inplace}.bash` unless they need to run independently as well
2) use separate variables, 3) make the "find upwards" logic more complex e.g. `$(while [ $PWD != / ] && [ ! -e $PWD/flake.nix ]; do cd ..; done; if [ $PWD = "/" ]; exit 1; else; echo $PWD; fi)`.

Besides if all we want is a `flox` binary that can be called from anywhere we can also make the wrapper more specific to always exec `FLOX_BIN`, and in doubt using a nix build is still an option -- likely the most reliable one in either case.

Since that's detail id prefer to go forward with this for now and discuss this usecase separately.